### PR TITLE
[Flow EVM] Improving evm integration tests

### PR DESCRIPTION
**Pull request by ramtinms**
closes: #5228

It also adds some utility methods for integration testing that can be used later to expand the integration tests.

### CODE REVIEW

File: fvm/evm/evm_test.go [hunk 1]
```diff
@@ -223,32 +433,148 @@ func TestCadenceOwnedAccountDeploy(t *testing.T) {
 					let cadenceOwnedAccount <- EVM.createCadenceOwnedAccount()
 					cadenceOwnedAccount.deposit(from: <-vault)
 
-					let address = cadenceOwnedAccount.deploy(
-						code: [],
-						gasLimit: 53000,
-						value: EVM.Balance(attoflow: 1230000000000000000)
+					let bal = EVM.Balance(0)
```
Comment (m-Peter) [inline fvm/evm/evm_test.go:436]:
```suggestion
					let bal = EVM.Balance(attoflow: 0)
```

Comment (m-Peter) [reply fvm/evm/evm_test.go:436]: This is a breaking change in Cadence 1.0. The named attribute is always required.

File: fvm/evm/evm_test.go [hunk 2]
```diff
@@ -223,32 +433,148 @@ func TestCadenceOwnedAccountDeploy(t *testing.T) {
...
+					let admin = getAuthAccount(%s)
+						.borrow<&FlowToken.Administrator>(from: /storage/flowTokenAdmin)!
+					let minter <- admin.createNewMinter(allowedAmount: 2.34)
+					let vault <- minter.mintTokens(amount: 2.34)
+					destroy minter
+
+					let cadenceOwnedAccount <- EVM.createCadenceOwnedAccount()
+					cadenceOwnedAccount.deposit(from: <-vault)
+
+					let bal = EVM.Balance(0)
```
Comment (m-Peter) [inline fvm/evm/evm_test.go:503]:
```suggestion
					let bal = EVM.Balance(attoflow: 0)
```

File: fvm/evm/evm_test.go [hunk 3]
```diff
@@ -430,6 +734,235 @@ func TestCadenceArch(t *testing.T) {
...
+func getFlowAccountBalance(
+	t *testing.T,
+	ctx fvm.Context,
+	vm fvm.VM,
+	snap snapshot.SnapshotTree,
+	address flow.Address,
+) uint64 {
+	code := []byte(fmt.Sprintf(
+		`
+		pub fun main(): UFix64 {
```
Comment (m-Peter) [inline fvm/evm/evm_test.go:871]:
```suggestion
		access(all) fun main(): UFix64 {
```

**Review by m-Peter — APPROVED** [review]
Very nice :clap: 
Left only a few comments, to make the Cadence changes compatible with C1.0.

File: fvm/evm/evm_test.go [hunk 4]
```diff
@@ -51,29 +106,133 @@ func TestEVMRun(t *testing.T) {
...
+					import EVM from %s
+
+					transaction(tx: [UInt8], coinbaseBytes: [UInt8; 20]){
+						prepare(account: AuthAccount) {
+							let coinbase = EVM.EVMAddress(bytes: coinbaseBytes)
+							let res = EVM.run(tx: tx, coinbase: coinbase)
+
+							assert(res.status == EVM.Status.failed, message: "unexpected status")
+							// ExecutionErrCodeExecutionReverted
+							assert(res.errorCode == 306, message: "unexpected error code")
```
Comment (sideninja) [inline fvm/evm/evm_test.go:163]: maybe:
```suggestion
							assert(res.errorCode == %d, message: "unexpected error code")
							// ...
							types.ExecutionErrCodeExecutionReverted
```

File: fvm/evm/evm_test.go [hunk 5]
```diff
@@ -223,32 +433,148 @@ func TestCadenceOwnedAccountDeploy(t *testing.T) {
...
+					let minter <- admin.createNewMinter(allowedAmount: 2.34)
+					let vault <- minter.mintTokens(amount: 2.34)
+					destroy minter
+
+					let cadenceOwnedAccount <- EVM.createCadenceOwnedAccount()
+					cadenceOwnedAccount.deposit(from: <-vault)
+
+					let bal = EVM.Balance(0)
+					bal.setFLOW(flow: 1.23)
+					let vault2 <- cadenceOwnedAccount.withdraw(balance: bal)
```
Comment (sideninja) [inline fvm/evm/evm_test.go:505]: nit: maybe you can add a line in Cadence that asserts the coa balance is `2.34-1.23`.

**Review by janezpodhostnik — APPROVED** [review]

**Comment by ramtinms** [comment]
I'll apply suggested these in my other PR https://github.com/onflow/flow-go/pull/5443

**Comment by ramtinms** [comment]
applied fixes to https://github.com/onflow/flow-go/pull/5443/commits/7ae2c70c38a198d94f3a1980d6e9fd912615812a

**Pull request merged**